pkg/network/http/testutil/testutil.go

```diff
@@ -47,35 +47,33 @@ func HTTPServer(t *testing.T, addr string, options Options) func() {
 		ReadTimeout:  time.Second,
 		WriteTimeout: time.Second,
 	}
+	srv.SetKeepAlivesEnabled(options.EnableKeepAlives)
 
-	listenFn := func() { _ = srv.ListenAndServe() }
+	listenFn := func() error {
+		ln, err := net.Listen("tcp", srv.Addr)
+		if err == nil {
+			go func() { _ = srv.Serve(ln) }()
```

Comment:
ah, nice! I didn't know the standard library server had a Listen/Serve methods :grimacing:
I guess you got rid of the timeout stuff because `Listen` blocks until the socket is up?

Reply:
Yes, exactly!